src/js/CheckboxTree.js

```diff
@@ -59,9 +56,11 @@ class CheckboxTree extends React.Component {
             parentClose: <span className="rct-icon rct-icon-parent-close" />,
             parentOpen: <span className="rct-icon rct-icon-parent-open" />,
             leaf: <span className="rct-icon rct-icon-leaf" />,
+            radioOff: <span className="rct-icon rct-icon-radio-off" />,
+            radioOn: <span className="rct-icon rct-icon-radio-on" />,
         },
         iconsClass: 'fa4',
-        id: null,
+        // id: null,
```

Comment:
`id` was mistakenly removed from `PropTypes`.

The props are checked for `id` in `getDerivedStateFromProps` and the state updated there so it is fully enabled. Only if the id is changed is the state updated. This keeps `TreeNodes` from re-rendering unnecessarily. I was having `id` cause them to re-render every time.

Note that also `icons` is set to `defaultProps.icons` in state and also checked in `getDerivedStateFromProps`.  This is done this way because in the previous code, `icons` was recreated as a prop to like this:
```
<TreeNode
      ...
      icons={{...defaultIcons, ...icons}}
      ...
>
```
This creates a new object for icons in every instance of `TreeNode` and causes re-rendering every time.  The way I've done it `icons` should only change when the input prop to `CheckTree` changes.